### ISSUE TITLE
SP601E stuck in 'Initializing' state

### ISSUE THREAD

**Issue by tobiastoft** [monty68/uniled#34]
Version: 1.03
Installed through HACS on Home Assistant OS 11.1
Home Assistant 2023.10.5

Device model: SP601E
BT Device name: SP601E

## Description
Uniled can detect the SP601E fine, but when I hit "Configure" it gets stuck in the "Initializing" stage. When inspecting the log it just repeats the pattern below over and over.

## Debug log

```text
2023-10-30 13:05:45.936 DEBUG (MainThread) [custom_components.uniled.config_flow] Setting discovery: <home_assistant_bluetooth.models.BluetoothServiceInfoBleak object at 0x7f760c786940>
2023-10-30 13:05:47.792 DEBUG (MainThread) [custom_components.uniled.config_flow] User Input: {'address': '00:55:44:52:2A:B4'}
2023-10-30 13:05:47.792 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 0
2023-10-30 13:05:47.792 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Init BLE Device (Model: SP601E)
2023-10-30 13:05:47.792 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Set model SP601E
2023-10-30 13:05:47.792 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 1
2023-10-30 13:05:47.792 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 2
2023-10-30 13:05:47.792 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] Device Destroyed
2023-10-30 13:05:47.793 DEBUG (MainThread) [custom_components.uniled.lib.classes] Channel 2 Destroyed
2023-10-30 13:05:47.793 DEBUG (MainThread) [custom_components.uniled.lib.classes] Channel 1 Destroyed
2023-10-30 13:05:47.793 DEBUG (MainThread) [custom_components.uniled.lib.classes] Channel 0 Destroyed
2023-10-30 13:05:47.799 DEBUG (MainThread) [custom_components.uniled] *** Setup UniLED BLE Device: 00:55:44:52:2A:B4: SP601E v1 - ({'device_class': 'ble', 'address': '00:55:44:52:2A:B4', 'model': 'SP601E'}), Service: <home_assistant_bluetooth.models.BluetoothServiceInfoBleak object at 0x7f75ed0717c0>
2023-10-30 13:05:47.800 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 0
2023-10-30 13:05:47.800 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Init BLE Device (Model: SP601E)
2023-10-30 13:05:47.800 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Set model SP601E
2023-10-30 13:05:47.800 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 1
2023-10-30 13:05:47.800 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 2
2023-10-30 13:05:47.801 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Update
2023-10-30 13:05:47.801 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Connecting; RSSI: -81
2023-10-30 13:05:48.593 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Connected
2023-10-30 13:05:48.593 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Read Characteristic: 0000ffe1-0000-1000-8000-00805f9b34fb (Handle: 13): Vendor specific
2023-10-30 13:05:48.593 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Write Characteristic: 0000ffe1-0000-1000-8000-00805f9b34fb (Handle: 13): Vendor specific
2023-10-30 13:05:48.594 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Subscribe to notifications
2023-10-30 13:05:48.624 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Sending 1 command(s)
2023-10-30 13:05:48.624 DEBUG (MainThread) [custom_components.uniled.lib.ble_retry] SP601E: Starting retry loop
2023-10-30 13:05:48.624 DEBUG (MainThread) [custom_components.uniled.lib.ble_retry] SP601E: Attempt 1
2023-10-30 13:05:48.624 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Sending command: aa2f00 
2023-10-30 13:05:48.625 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Awaiting status notification
2023-10-30 13:05:50.626 WARNING (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Timeout waiting for status notification
2023-10-30 13:05:50.626 DEBUG (MainThread) [custom_components.uniled.coordinator] Finished fetching SP601E data in 2.825 seconds (success: False)
2023-10-30 13:05:50.626 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Stop
2023-10-30 13:05:50.626 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Disconnecting from device
2023-10-30 13:05:50.627 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Stopped notifications from device
2023-10-30 13:05:50.728 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Device disconnected (expected: True); RSSI: -81
2023-10-30 13:05:50.729 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Disconnected from device
2023-10-30 13:05:50.730 WARNING (MainThread) [homeassistant.config_entries] Config entry 'SP601E' for uniled integration not ready yet: SP601E: Data Update failed; Retrying in background
2023-10-30 13:05:55.854 DEBUG (MainThread) [custom_components.uniled] *** Setup UniLED BLE Device: 00:55:44:52:2A:B4: None v1 - ({'device_class': 'ble', 'address': '00:55:44:52:2A:B4', 'model': 'SP601E'}), Service: <home_assistant_bluetooth.models.BluetoothServiceInfoBleak object at 0x7f75ea9b3bc0>
2023-10-30 13:05:55.855 DEBUG (MainThread) [custom_components.uniled.lib.classes] 00:55:44:52:2A:B4: Init Channel 0
2023-10-30 13:05:55.855 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] None: Init BLE Device (Model: SP601E)
2023-10-30 13:05:55.855 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] 00:55:44:52:2A:B4: Set model SP601E
2023-10-30 13:05:55.855 DEBUG (MainThread) [custom_components.uniled.lib.classes] 00:55:44:52:2A:B4: Init Channel 1
2023-10-30 13:05:55.855 DEBUG (MainThread) [custom_components.uniled.lib.classes] 00:55:44:52:2A:B4: Init Channel 2
2023-10-30 13:05:55.855 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] 00:55:44:52:2A:B4: Update
2023-10-30 13:05:55.857 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] 00:55:44:52:2A:B4: Connecting; RSSI: -74
2023-10-30 13:05:56.462 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Connected
2023-10-30 13:05:56.463 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Read Characteristic: 0000ffe1-0000-1000-8000-00805f9b34fb (Handle: 13): Vendor specific
2023-10-30 13:05:56.463 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Write Characteristic: 0000ffe1-0000-1000-8000-00805f9b34fb (Handle: 13): Vendor specific
2023-10-30 13:05:56.463 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Subscribe to notifications
2023-10-30 13:05:56.583 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Sending 1 command(s)
2023-10-30 13:05:56.583 DEBUG (MainThread) [custom_components.uniled.lib.ble_retry] SP601E: Starting retry loop
2023-10-30 13:05:56.583 DEBUG (MainThread) [custom_components.uniled.lib.ble_retry] SP601E: Attempt 1
2023-10-30 13:05:56.583 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Sending command: aa2f00 
2023-10-30 13:05:56.584 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Awaiting status notification
2023-10-30 13:05:58.586 WARNING (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Timeout waiting for status notification
2023-10-30 13:05:58.586 DEBUG (MainThread) [custom_components.uniled.coordinator] Finished fetching 00:55:44:52:2A:B4 data in 2.731 seconds (success: False)
2023-10-30 13:05:58.586 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Stop
2023-10-30 13:05:58.586 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Disconnecting from device
2023-10-30 13:05:58.587 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Stopped notifications from device
2023-10-30 13:05:58.716 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Device disconnected (expected: True); RSSI: -83
2023-10-30 13:05:58.716 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Disconnected from device
2023-10-30 13:05:58.719 DEBUG (MainThread) [custom_components.uniled] *** Setup UniLED BLE Device: 00:55:44:52:2A:B4: SP601E v1 - ({'device_class': 'ble', 'address': '00:55:44:52:2A:B4', 'model': 'SP601E'}), Service: <home_assistant_bluetooth.models.BluetoothServiceInfoBleak object at 0x7f75ecfff1c0>
2023-10-30 13:05:58.719 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 0
2023-10-30 13:05:58.720 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Init BLE Device (Model: SP601E)
2023-10-30 13:05:58.720 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Set model SP601E
2023-10-30 13:05:58.720 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 1
2023-10-30 13:05:58.720 DEBUG (MainThread) [custom_components.uniled.lib.classes] SP601E: Init Channel 2
2023-10-30 13:05:58.721 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Update
2023-10-30 13:05:58.721 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Connecting; RSSI: -92
2023-10-30 13:05:59.229 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Connected
2023-10-30 13:05:59.230 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Read Characteristic: 0000ffe1-0000-1000-8000-00805f9b34fb (Handle: 13): Vendor specific
2023-10-30 13:05:59.230 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Write Characteristic: 0000ffe1-0000-1000-8000-00805f9b34fb (Handle: 13): Vendor specific
2023-10-30 13:05:59.230 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Subscribe to notifications
2023-10-30 13:05:59.330 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Sending 1 command(s)
2023-10-30 13:05:59.331 DEBUG (MainThread) [custom_components.uniled.lib.ble_retry] SP601E: Starting retry loop
2023-10-30 13:05:59.331 DEBUG (MainThread) [custom_components.uniled.lib.ble_retry] SP601E: Attempt 1
2023-10-30 13:05:59.331 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Sending command: aa2f00 
2023-10-30 13:05:59.332 DEBUG (MainThread) [custom_components.uniled.lib.ble_device] SP601E: Awaiting status notification

```

**Comment by monty68** [comment]
Fixed in v1.04

**Comment by tobiastoft** [comment]
It works flawlessly, thank you!
